core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -612,12 +617,26 @@ private[log] class Cleaner(val id: Int,
       if (outputBuffer.position() > 0) {
         outputBuffer.flip()
         val retained = MemoryRecords.readableRecords(outputBuffer)
+        val baseOffsetOfLog = dest.baseOffset
+        val firstOffsetToAppend = retained.batches.iterator.next.baseOffset
```

Comment:
`baseOffset` is an expensive operation for the old message format. Since we are only using this in logging, maybe we could leave it out?